src/openrct2/entity/Guest.cpp

```diff
@@ -1804,6 +1804,52 @@ void Guest::OnExitRide(Ride& ride)
     ride.window_invalidate_flags |= RIDE_INVALIDATE_RIDE_CUSTOMER;
 }
 
+
+void Guest::SendToRide(RideId rideId)
+{
+    if (rideId.IsNull())
+        return;
+    Ride* ride = GetRide(rideId);
+    if (ride == nullptr)
+        return;
+    if (x == LOCATION_NULL)
+        return;
+
```

Comment:
This function does not seem to check if the guest is actually in the park.

Reply:
this line was copypasted from Guest::PickRideToGoOn(), which was what I based all this code on. I'm not sure what it does but left it since it didn't obviously hurt anything. 

the guest is checked to be in the park in the PeepState check.